sonar-plugin-api/src/main/java/org/sonar/api/batch/rule/Rule.java

```diff
@@ -51,19 +50,19 @@
   RuleStatus status();
 
   /**
-   * Sub characteristic key.
-   *
    * @since 4.3
+   * @deprecated since 5.2 as any computation of data are moved to server's Compute Engine. Calling this method throws an exception.
    */
   @CheckForNull
+  @Deprecated
   String debtSubCharacteristic();
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)